scheduler/scheduler-server/src/test/java/org/ow2/proactive/utils/PCAProxyRuleTest.java

```diff
@@ -0,0 +1,115 @@
+        String expected = "/cloud-automation-service/services/93/endpoints/jupyterlab-endpoint2/lab/workspaces/auto-s/lab";
+        runTest(requestPath, HttpMethod.GET, referer, expected);
+    }
+
+    @Test
+    public void matchAndApplyWithCloudAutomationReferer() throws IOException {
+        String requestPath = "/lab/workspaces/";
+        String referer = "http://myserver.com:8080/cloud-automation-service/services/93/endpoints/jupyterlab-endpoint2/lab?clone";
+        String expected = "/cloud-automation-service/services/93/endpoints/jupyterlab-endpoint2/lab/workspaces/";
+        runTest(requestPath, HttpMethod.GET, referer, expected);
+    }
+
+    @Test
+    public void matchAndApplyWithCloudAutomationRefererPostMethod() throws IOException {
+        String requestPath = "/lab/workspaces/";
+        String referer = "http://myserver.com:8080/cloud-automation-service/services/93/endpoints/jupyterlab-endpoint2/lab?clone";
+        String expected = "/cloud-automation-service/services/93/endpoints/jupyterlab-endpoint2/lab/workspaces/";
+        runTest(requestPath, HttpMethod.POST, referer, expected);
+    }
+
+    @Test
+    public void matchAndApplyWithCloudAutomationRefererAndQuery() throws IOException {
+        String requestPath = "/lab/workspaces/?param1=value1&param2=value2";
+        String referer = "http://myserver.com:8080/cloud-automation-service/services/93/endpoints/jupyterlab-endpoint2/lab?clone";
+        String expected = "/cloud-automation-service/services/93/endpoints/jupyterlab-endpoint2/lab/workspaces/?param1=value1&param2=value2";
+        runTest(requestPath, HttpMethod.GET, referer, expected);
+    }
+
+    private void runTest(String requestPath, HttpMethod method, String referer, String expectedTarget)
+            throws IOException {
+        PCAProxyRule rule = new PCAProxyRule();
+        Request request = Mockito.mock(Request.class);
+        HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+        String target = requestPath;
+        HttpURI requestUri = new HttpURI(target);
+        when(request.getMethod()).thenReturn(method.asString());
+        when(request.getHeader(HttpHeaders.REFERER)).thenReturn(referer);
+        when(request.getUri()).thenReturn(requestUri);
+        ArgumentCaptor<String> redirectString = ArgumentCaptor.forClass(String.class);
+        when(response.encodeRedirectURL(anyString())).then(returnsFirstArg());
+        String newTarget = rule.matchAndApply(target, request, response);
+        Assert.assertEquals(expectedTarget, newTarget);
+        if (HttpMethod.GET.is(method.asString())) {
+            verify(response, times(1)).sendRedirect(redirectString.capture());
+            Assert.assertEquals(expectedTarget, redirectString.getValue());
+        } else {
+            verify(response, times(0)).sendRedirect(redirectString.capture());
+        }
+
+    }
+
+    @Test
+    public void applyURI() {
```

Comment:
Is this test supposed to be empty ?

Reply:
sorry, this is a mistake (and I will remove it probably)